src/bsoncxx/oid.cpp

```diff
@@ -34,7 +34,11 @@ oid::oid() {
 
 const oid::init_tag_t oid::init_tag{};
 
-oid::oid(init_tag_t) : oid::oid() {}
+oid::oid(init_tag_t) : oid::oid(init_tag_deprecated) {}
+
+const oid::init_tag_t_deprecated oid::init_tag_deprecated{};
+
+oid::oid(init_tag_t_deprecated) : oid::oid() {}
```

Comment:
Is this a new deprecation?

Reply:
No - it lets you invoke the deprecated constructor without triggering the deprecation warning.